kernel/kernel-api/src/main/java/io/delta/kernel/expressions/CollationIdentifier.java

```diff
@@ -0,0 +1,38 @@
+package io.delta.kernel.expressions;
+
+import java.util.Optional;
+
+public class CollationIdentifier {
+  public static final String PROVIDER_KERNEL = "KERNEL";
```

Comment:
i guess this is not clear, does it make sense to have `"spark"` as the provider if we want to mimic the spark/delta behavior? @vkorukanti